src/home_robot/home_robot/utils/morphology.py

```diff
@@ -49,3 +50,103 @@ def binary_closing(binary_image, kernel):
 
 def binary_denoising(binary_image, kernel):
     return binary_opening(binary_closing(binary_image, kernel), kernel)
+
+
+def get_edges(mask: torch.Tensor, threshold: float = 0.5) -> torch.Tensor:
+    """Extract edges from a torch tensor."""
+
+    mask = mask.float()
+
+    # Define the Sobel filter kernels
+    sobel_x = torch.tensor([[-1, 0, 1], [-2, 0, 2], [-1, 0, 1]], dtype=torch.float32)
+    sobel_y = torch.tensor([[-1, -2, -1], [0, 0, 0], [1, 2, 1]], dtype=torch.float32)
+
+    # Calculate padding for convolution to preserve the original size
+    # Sobel x and y operators are the same size
+    padding_x = sobel_x.size(0) // 2
+    padding_y = sobel_x.size(1) // 2
+
+    # Apply Sobel filter to detect edges in x and y directions
+    edges_x = F.conv2d(
+        mask.unsqueeze(0).unsqueeze(0),
+        sobel_x.unsqueeze(0).unsqueeze(0),
+        padding=(padding_x, padding_y),
+    )
+    edges_y = F.conv2d(
+        mask.unsqueeze(0).unsqueeze(0),
+        sobel_y.unsqueeze(0).unsqueeze(0),
+        padding=(padding_x, padding_y),
+    )
+
+    # Combine x and y edge responses to get the magnitude of edges
+    edges = torch.sqrt(edges_x**2 + edges_y**2)
+    edges = edges[0, 0] > threshold
+    assert (
+        edges.shape == mask.shape
+    ), "something went wrong when computing padding, most likely - shape not preserved"
+    return edges
+
+
+def expand_mask(mask: torch.Tensor, radius: int, threshold: float = 0.5):
+    """Expand a mask by some radius in pytorch"""
+
+    # Needs to be converted to a float to work
+    mask = mask.float()
+
+    # Create a disk-shaped structuring element
+    x, y = torch.meshgrid(
+        torch.arange(-radius, radius + 1), torch.arange(-radius, radius + 1)
+    )
+    selem = (x**2 + y**2 <= radius**2).to(torch.float32)
+
+    # Calculate padding for convolution to preserve the original size
+    padding_x = selem.size(0) // 2
+    padding_y = selem.size(1) // 2
+
+    # Apply binary dilation to expand the mask
+    expanded_mask = F.conv2d(
+        mask.unsqueeze(0).unsqueeze(0),
+        selem.unsqueeze(0).unsqueeze(0),
+        padding=(padding_x, padding_y),
+    )
+
+    # Binarize the expanded mask (optional)
+    expanded_mask = (expanded_mask > 0).to(torch.float32)
+    expanded_mask = expanded_mask[0, 0] > threshold
+    assert (
+        expanded_mask.shape == mask.shape
+    ), "something went wrong when computing padding, most likely - shape not preserved"
```

Comment:
Nice, informative error message. I like adding the shapes to the output string -- this usually saves me from rerunning the command